### ISSUE TITLE
Add `M3` to internal datasets

### ISSUE THREAD

**Issue by ostreech1997** [etna-team/etna#84]
### 🚀 Feature Request

We want to add `M3` dataset to internal datasets.

### Proposal

- Add `M3` with different seasonalities to `datasets_dict `:
```Python
datasets_dict = {
    "m3_monthly": {
        "get_dataset_function": partial(get_m3_dataset, dataset_freq="monthly"),
        "freq": "M",
        "parts": ("train", "test", "full")
    },
    "m3_quarterly": {
        "get_dataset_function": partial(get_m3_dataset, dataset_freq="quarterly"),
        "freq": "Q",
        "parts": ("train", "test", "full")
    },
    "m3_yearly": {
        "get_dataset_function": partial(get_m3_dataset, dataset_freq="yearly"),
        "freq": "Y",
        "parts": ("train", "test", "full")
    },
    "m3_other": {
        "get_dataset_function": partial(get_m3_dataset, dataset_freq="other"),
        "freq": "Q",
        "parts": ("train", "test", "full")
    },
}
```
- Add `get_m3_dataset` function for loading [M3 dataset](https://forvis.github.io/datasets/m3-data/). We have to add dataset description and reference links to docstring: [initial source link](http://forecasters.org/resources/time-series-data/m3-competition/), [source that we use](https://forvis.github.io/datasets/m3-data/)). 
```Python
def get_m3_dataset(dataset_path: Path, dataset_freq: str) -> None:
```
*Important notion*
We have to create custom timestamps, because `TSDataset` can't work with series that have different last dates. We should also save real timestamps as exogenous data.

### Test cases

- Check `M3` datasets (number of segments, min/max dates).
- Add tests that check for the existence of exogenous data.

### Additional context

See https://github.com/etna-team/etna/pull/83 as an example.

**Comment by egoriyaa** [comment]
Closed by #91

**Comment by egoriyaa** [comment]
Closed by #91
